python/ray/tests/test_runtime_env.py

```diff
@@ -99,12 +112,54 @@ def test_two_node_uri(two_node_cluster, working_dir):
         runtime_env.create_project_package(working_dir, [], tmp_file.name)
         runtime_env.push_package(pkg_uri, tmp_file.name)
         runtime_env = f"""{{ "working_dir_uri": "{pkg_uri}" }}"""
-    script = driver_script.format(
-        redis_address=redis_address,
-        working_dir=working_dir,
-        runtime_env=runtime_env)
+        execute_statement = "print(sum(ray.get([run_test.remote()] * 1000)))"
+    script = driver_script.format(**locals())
+    out = run_string_as_driver(script)
+    assert out.strip().split()[-1] == "1000"
+    from ray._private.runtime_env import PKG_DIR
+    assert len(list(Path(PKG_DIR).glob("*.zip"))) == 0
+
+
+@unittest.skipIf(sys.platform == "win32", "Fail to create temp dir.")
+def test_regular_actors(ray_start_cluster_head, working_dir):
+    cluster = ray_start_cluster_head
+    redis_address = cluster.address
+    runtime_env = f"""{{  "working_dir": "{working_dir}" }}"""
+    execute_statement = """
+test_actor = TestActor.options(name="test_actor").remote()
+print(sum(ray.get([test_actor.one.remote()] * 1000)))
+"""
+    script = driver_script.format(**locals())
     out = run_string_as_driver(script)
     assert out.strip().split()[-1] == "1000"
+    from ray._private.runtime_env import PKG_DIR
+    assert len(list(Path(PKG_DIR).glob("*.zip"))) == 0
+
+
+@unittest.skipIf(sys.platform == "win32", "Fail to create temp dir.")
+def test_detached_actors(ray_start_cluster_head, working_dir):
+    cluster = ray_start_cluster_head
+    redis_address = cluster.address
+    runtime_env = f"""{{  "working_dir": "{working_dir}" }}"""
+    execute_statement = """
+test_actor = TestActor.options(name="test_actor", lifetime="detached").remote()
+print(sum(ray.get([test_actor.one.remote()] * 1000)))
+"""
+    script = driver_script.format(**locals())
+    out = run_string_as_driver(script)
+    assert out.strip().split()[-1] == "1000"
+    from ray._private.runtime_env import PKG_DIR
+    # It's a detached actors, so it should still be there
+    assert len(list(Path(PKG_DIR).glob("*.zip"))) == 1
+    pkg = list(Path(PKG_DIR).glob("*.zip"))[0]
+    import sys
+    sys.path.insert(0, str(pkg))
+    test_actor = ray.get_actor("test_actor")
```

Comment:
For the driver which wants to use the detached actor, it also needs to have the env as that actor. I hope it's a reasonable assumption. @ericl  @edoakes 

Here for testing, it needs to set manually, but in the normal use case, if we set the runtime env correctly, it'll just work.

Reply:
This is fine for now, but we need to support setting per-task/actor runtime_env in the future.